build-scripts/cassandra-artifacts.sh

```diff
@@ -38,11 +40,18 @@ python -m pip install "setuptools<45" Sphinx sphinx_rtd_theme
 
 # Setup JDK
 java_version=$(java -version 2>&1 | awk -F '"' '/version/ {print $2}' | awk -F. '{print $1}')
-if [ "$java_version" -ge 11 ]; then
+if [ "$java_version" -ge 17 ]; then
+    java_version="17"
+    export CASSANDRA_USE_JDK17=true
+    if ! grep -q "java.version.17" build.xml ; then
+        echo "Invalid JDK17 not supported against ${cassandra_version}"
+        exit 1
+    fi
+elif [ "$java_version" -ge 11 ]; then
     java_version="11"
     export CASSANDRA_USE_JDK11=true
-    if ! grep -q CASSANDRA_USE_JDK11 build.xml ; then
-        echo "Skipping build. JDK11 not supported against $(grep 'property\s*name=\"base.version\"' build.xml |sed -ne 's/.*value=\"\([^"]*\)\".*/\1/p')"
+    if ! grep -q "java.version.11" build.xml ; then
```

Comment:
switching these greps over is important to do before [CASSANDRA-18133](https://issues.apache.org/jira/browse/CASSANDRA-18133)